File: app/src/main/java/co/fddittmar/planbuilder/contracts/MainContract.java
package co.fddittmar.planbuilder.contracts;

import java.util.List;

import co.fddittmar.planbuilder.data.model.Program;

/**
 * Contract for the Main activity. It has an interface for the View and Presenter.
 */

public class MainContract {

    public interface View {
        void hideNewProgramText();
        void showNewProgramText();
        void openNewProgramActivity();
        void openProgramDetailActivity(Program program);
        void refreshAdapter();
        void newProgramAdded();
    }

    public interface Actions {

        List<Program> getAllPrograms();
        void checkNewProgramText(List<Program> programList);
        void onAddBtnClicked();
        void onItemProgramClicked(Program program);
        void checkActivityResult(int resultCode);
        void deleteProgram(long program_id);
    }
}

File: app/src/main/java/co/fddittmar/planbuilder/contracts/MainContract.kt
package co.fddittmar.planbuilder.contracts

import co.fddittmar.planbuilder.data.model.Program

/**
 * Contract for the Main activity. It has an interface for the View and Presenter.
 */

class MainContract {

    interface View {
        fun hideNewProgramText()
        fun showNewProgramText()
        fun openNewProgramActivity()
        fun openProgramDetailActivity(program: Program)
        fun refreshAdapter()
        fun newProgramAdded()
    }

    interface Actions {

        val allPrograms: ArrayList<Program>
        fun checkNewProgramText(programList: List<Program>)
        fun onAddBtnClicked()
        fun onItemProgramClicked(program: Program)
        fun checkActivityResult(resultCode: Int)
        fun deleteProgram(program_id: Long)
    }
}

